### ISSUE TITLE
4th order ode: dsolve gives explicit solution selectively

### ISSUE THREAD

**Issue by ChenguangZhang** [sympy/sympy#15428]
(version 1.3)
Hi all,
Here is what I mean, the code below gives the fully, explicit solution without problem. By explicit solution I mean there is no RootOf(sth) term in the solution.
```
f = Function('f');
ode = Eq(diff(f(t), t, 4) + diff(f(t), t, 2) + f(t), 0)
dsolve(ode, f(t))
```
But if I change one coefficient of the ODE
```
ode = Eq(diff(f(t), t, 4) + 4*diff(f(t), t, 2) + f(t), 0)
```
The solution given by dsolve becomes implicit: it becomes a sum of C_i*e^{CRootOf(x^4+4*x^2+1,i)}, i from 0 to 3. This is surprising because sympy _can_ solve the root without problem, for example, the command below 
```
solve(Eq(x**4 + 4*x*x + 1, 0), x)
```
gives all four roots.

I don't know if I can call this as a bug, but at least this is inconsistent behavior.

**Comment by asmeurer** [comment]
The problem is that `rootof(x**4 + 4*x*x + 1, 0)` doesn't automatically simplify. `solve` is not used because it doesn't return RootOf instances, and always wants to return all the solutions even when the roots of the characteristic equation cannot be expressed as radicals.

**Comment by asmeurer** [comment]
Even `CRootOf(x**4 + 4*x*x + 1, 0, radicals=True)` doesn't work, even though it should according to the docstring. So this is clearly a bug with RootOf not accessing the polynomial solvers properly.

**Comment by ChenguangZhang** [comment]
As a side note, right before reporting this issue, I was using sympy version 1.0. 

In that case, dsolve-ing the second ode Eq(diff(f(t), t, 4) + 4*diff(f(t), t, 2) + f(t), 0) would hang several minutes before I kill the command, whereas dsolve-ing the first ode returns the correct solution instantly. So the behavior is even more inconsistent. I agree with @asmeurer that this is a bug; a bigger one in version 1.0, and a smaller one in version 1.3.

An observation
- The roots of x**4 + x**2 + 1 ==0 are complex
- The roots of x**4 + 4*x**2 + 1 ==0 are complex but also **purely imaginary**
Maybe this is the cause?

**Comment by oscarbenjamin** [comment]
This issue is now fixed on master since #15408 is merged.
